pkg/api/addtask.go

```diff
@@ -0,0 +1,88 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/mariya-goncharenko/go_final_project_my/pkg/db"
+)
+
+// addTaskHandler обрабатывает POST-запросы на создание задачи
+func addTaskHandler(w http.ResponseWriter, r *http.Request) {
+	var task db.Task
+
+	// Чтение JSON из тела запроса
+	err := json.NewDecoder(r.Body).Decode(&task)
+	if err != nil {
+		writeJson(w, map[string]string{"error": fmt.Sprintf("ошибка декодирования JSON: %v", err)})
+		return
+	}
+
+	// Проверка обязательного поля title
+	if task.Title == "" {
+		writeJson(w, map[string]string{"error": "Не указан заголовок задачи"})
+		return
+	}
+
+	// Проверка даты и правила повторения
+	err = checkDate(&task)
+	if err != nil {
+		writeJson(w, map[string]string{"error": err.Error()})
+		return
+	}
+
+	// Добавляем задачу в БД
+	id, err := db.AddTask(&task)
+	if err != nil {
+		writeJson(w, map[string]string{"error": fmt.Sprintf("ошибка записи в БД: %v", err)})
+		return
+	}
+
+	// Возвращаем id добавленной задачи
+	writeJson(w, map[string]string{"id": fmt.Sprintf("%d", id)})
+}
+
+// checkDate проверяет дату и правило повторения и изменяет дату задачи, если необходимо
+func checkDate(task *db.Task) error {
+	now := time.Now()
+	now = time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location())
+	layout := "20060102"
+
+	// Если дата не указана — подставляем сегодняшнюю
+	if task.Date == "" {
+		task.Date = now.Format(layout)
+	}
+
+	// Проверка формата даты
+	t, err := time.Parse(layout, task.Date)
+	if err != nil {
+		return fmt.Errorf("неверный формат даты")
+	}
+
+	// Если указано правило повторения — проверяем и вычисляем следующую дату
+	if len(task.Repeat) > 0 {
+		next, err := CalculateNextDate(now, task.Date, task.Repeat)
+		if err != nil {
+			return fmt.Errorf("неверное правило повторения: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
			return fmt.Errorf("неверное правило повторения: %w", err)
```

Reply:
Исправила